ShoppingCart.php

```diff
@@ -279,6 +281,34 @@ public function getCost($withDiscount = false)
     }
 
     /**
+     * Return full cart shipping cost
+     * @param $withDiscount
+     * @return int
+     */
+    public function getShippingCost($withDiscount = false)
```

Comment:
Why shipping cost must be in it's own separate method? It is possible to add shipping cost value during general computations using EVENT_COST_CALCULATION. If shipping cost should be based on the calculated cart cost, than probably it is possible to add event EVENT_AFTER_COST_CALCULATION and evaluate value of the cost there